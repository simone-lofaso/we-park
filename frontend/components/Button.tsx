import { ReactNode } from 'react'
import { Pressable, PressableProps, StyleSheet, View } from 'react-native'
import { Text } from './Text'

type ButtonProps = {
  text: string
} & PressableProps

export const Button = ({ children, text, ...props }: ButtonProps) => {

  return (
    <Pressable style={styles.button} {...props}>
      <Text style={styles.text}>{text}</Text>
      {children as ReactNode}
    </Pressable>
  )
}

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: '#45B499',
    borderRadius: 20,
    flexDirection: 'row',
    height: 60,
    justifyContent: 'center',
    width: '80%'
  },
  text: {
    fontSize: 30
  }
})