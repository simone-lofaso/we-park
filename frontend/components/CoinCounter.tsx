import { Image, StyleSheet, Text, View } from 'react-native';
import Token from '../assets/images/token.png';
import type { StyleProp, ViewStyle } from 'react-native';

export type CoinCounterProps = {
  tokens: number;
  style?: ViewStyle;
};

export function CoinCounter({ tokens, style }: CoinCounterProps) {
  return (
    <View style={{ ...style, ...styles.container }}>
      <Image source={Token} style={styles.image} />
      <Text style={styles.tokenAmount}>{tokens}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  image: {
    aspectRatio: 1,
    borderRadius: 100,
    display: 'flex',
    flex: 1,
    height: '12%',
    width: '12%',
  },
  tokenAmount: {
    color: 'white',
    flex: 6,
    fontFamily: 'Fredoka-One',
    fontSize: 30,
    marginLeft: 10,
  },
});
