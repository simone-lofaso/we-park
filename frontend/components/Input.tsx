import { StyleSheet, TextInput, TextInputProps } from 'react-native';
import { width } from '../constants/Dimensions';

export const Input = (props: TextInputProps) => {
  return <TextInput style={styles.input} {...props} />;
};

const styles = StyleSheet.create({
  input: {
    borderRadius: 20,
    backgroundColor: 'white',
    fontSize: 30,
    margin: 10,
    textAlign: 'center',
    width: 0.8 * width,
  },
});
