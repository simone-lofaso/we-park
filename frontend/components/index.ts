export * from './AuthForm';
export * from './Input';
export * from './Text';
export * from './Button';
export * from './BackButton';
