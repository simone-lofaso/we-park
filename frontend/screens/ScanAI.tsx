import { Image, SafeAreaView, StyleSheet, Text } from 'react-native';
import { green } from '../constants/Colors';
import * as FileSystem from 'expo-file-system';
import Splash from '../assets/images/splash.png';
import { RootStackScreenProps } from '../types';

export default function ScanAI({
  navigation,
  route,
}: RootStackScreenProps<'ScanAI'>) {
  const { Picture } = route.params;
  return (
    <SafeAreaView style={styles.container}>
      <Image
        source={{ uri: Picture }}
        style={{ width: '100%', height: '100%' }}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: green,
    justifyContent: 'center',
    height: '100%',
    padding: 20,
    width: '100%',
  },
});
