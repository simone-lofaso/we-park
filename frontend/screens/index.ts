export { default as Home } from './Home';
export { default as Login } from './Login';
export { default as Map } from './Map';
export { default as Register } from './Register';
export { default as ScanAI } from './ScanAI';
export { default as Profile } from './Profile';
