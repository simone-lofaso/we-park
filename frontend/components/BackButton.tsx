import { RootStackScreenProps } from '../types';
import { Button, ButtonProps } from './Button';

type BackButtonProps = {
  navigation: RootStackScreenProps<any>['navigation'];
} & Omit<ButtonProps, 'text'>;

export const BackButton = ({ navigation, ...props }: BackButtonProps) => {
  return <Button {...props} text='Back' onPress={() => navigation.pop(1)} />;
};
