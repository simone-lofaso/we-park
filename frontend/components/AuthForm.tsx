import { useState } from 'react';
import { Button, TextInputProps, StyleSheet, Text, View } from 'react-native';
import { doLogin, doRegister } from '../api';
import { RootStackScreenProps } from '../types';
import { Input } from './Input';

type AuthFormProps = {
  formType: 'Login' | 'Register';
  navigation: RootStackScreenProps<'Login' | 'Register'>['navigation'];
};

export const AuthForm = ({ formType, navigation }: AuthFormProps) => {
  const onPress = () => {
    if (formType === 'Login') {
      doLogin(form, navigation as RootStackScreenProps<'Login'>['navigation']);
    } else {
      doRegister(
        form,
        navigation as RootStackScreenProps<'Register'>['navigation']
      );
    }
  };

  const other = formType === 'Login' ? 'Register' : 'Login';

  const [form, setForm] = useState<{ email: string; password: string }>({
    email: '',
    password: '',
  });

  return (
    <View style={styles.container}>
      <Text style={styles.title}>WeParkâ„¢</Text>
      <Text style={styles.label}>{formType}</Text>
      <Input onChangeText={(text) => setForm({ ...form, email: text })} />
      <Text style={styles.label}>Email</Text>
      <Input
        secureTextEntry
        onChangeText={(text) => setForm({ ...form, password: text })}
      />
      <Text style={styles.label}>Password</Text>
      <Button title='Submit' onPress={onPress} />
      <Button
        title={`Go to ${other}`}
        onPress={async () => {
          navigation.navigate(`${other}`);
        }}
      />
      {/*Debugging only, delete later*/}
      <Button
        title={`Go to TakePicture (Debugging)`}
        onPress={async () => {
          navigation.navigate(`TakePicture`);
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    color: 'white',
    fontFamily: 'FredokaOne',
    fontSize: 75,
  },
  label: {
    color: 'white',
    fontFamily: 'FredokaOne',
    fontSize: 25,
    marginBottom: 20,
  },
  container: {
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
  },
  button: {
    margin: 10,
  },
});
