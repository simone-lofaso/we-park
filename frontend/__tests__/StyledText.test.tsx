import { create } from 'react-test-renderer';
import { MonoText } from '../components/StyledText';
import {ScanAI} from '../screens/ScanAI';

it(`renders correctly`, () => {
  const tree = create(<ScanAI/>).toJSON();
  expect(tree).toMatchSnapshot();
});
