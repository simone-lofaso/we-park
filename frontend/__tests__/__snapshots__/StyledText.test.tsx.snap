// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text>
  Hello world on purpoaw
</Text>
`;
